source/js/classes/Timer.js

```diff
@@ -34,36 +34,47 @@ class Timer {
    */
   startTimer() {
     return new Promise((resolve) => {
-      // Count down immediately
-      // This is where the timer callbacks a function every second
       const resolveMessage = 'Timer Finished';
-      if (this.callbackEverySecond !== null) {
-        this.callbackEverySecond(this.minutes, this.seconds);
-      }
-      if (this.seconds === 0 && this.minutes !== 0) {
-        this.minutes -= 1;
-        this.seconds = 60;
-      } else if (this.seconds === 0 && this.minutes === 0) {
+      if (this.seconds === 0 && this.minutes === 0) {
         resolve(resolveMessage);
       }
-      this.seconds -= 1;
+
       // Keep counting down until it reaches 0 mins 0 seconds (inclusive)
+      // Start at this.seconds - 1 to start counting down right away
+      const end = Math.floor(Date.now() / 1000) + (this.minutes * 60) + (this.seconds - 1);
+
+      // Start counter once because setInterval runs code only after timeout
+      this.timer(end);
       const countdown = setInterval(() => {
-        // This is where the timer callbacks a function every second
-        if (this.callbackEverySecond !== null) {
-          this.callbackEverySecond(this.minutes, this.seconds);
-        }
-        if (this.seconds === 0 && this.minutes !== 0) {
-          this.minutes -= 1;
-          this.seconds = 60;
-        } else if (this.seconds === 0 && this.minutes === 0) {
+        // End timer
+        if (this.seconds === 0 && this.minutes === 0) {
```

Comment:
Should change this to <= 0 because the timer can skip over 0 if performance is throttled, resulting in timer continuing infinitely with negative values

Reply:
Added in